homeassistant/components/hassio/backup.py

```diff
@@ -554,11 +558,22 @@ def on_job_progress(data: Mapping[str, Any]) -> None:
                 )
                 return
 
-            on_progress(
-                RestoreBackupEvent(
-                    reason="", stage=None, state=RestoreBackupState.COMPLETED
+            restore_errors = data.get("errors", [])
+            if restore_errors:
+                _LOGGER.warning("Restore backup failed: %s", restore_errors)
+                on_progress(
+                    RestoreBackupEvent(
+                        reason="unknown_error",
```

Comment:
Isn't the reason in the restore errors list?

Reply:
Yes, but it's not a well defined list of possible errors from supervisor. Also, we receive a list of errors, not just one.
In follow-up PRs, we could add handling of some known errors though.